### PR TITLE
cleanups and fixes to project path

### PR DESCRIPTION
**Pull request by niosus**
- switch to using sublime-provided way of getting current variables
- switch to absolute project path (should work on Windows too, but not tested)
- replace all by-hand path creation by using `os.path` way where found
- add verbose flag to enable showing found folders (like project, current file)
- add an option to include the file's parent folder to include folders
- add a new menu entry that allows the user to create a user settings file which will not be overwritten my an upgrade to the new version. The user has to populate it himself upon package installation. This should close issue #15 
- remove some annoying trailing spaces

The include paths relative to the project have not worked for me, so I have rewritten that part to provide absolute path. Also, sublime module provides us with paths for everything we care about, like project folder or current file path. 

I have also added a option to show the paths that the code has guessed with the "verbose" flag. The flag is false by default.

Another added option is to add the parent folder of current file to the include paths as I usually code in packages and the include paths in every file is relative to the package parent folder. There is a flag in setting to enable/disable this. The flag is false by default.

+ some cleanups like replacing all places where the paths were constructed manually with the os.path way. This should be cross platform, but I haven't checked it on windows. (I am on linux). 

Feel free to criticize this and tell me what you think. I am happy to include the suggestions you might have.

Overall, this is the best package for clang based completion and I thank you for your work :+1:

### CODE REVIEW

**Comment by pl-ca** [comment]
This is an awesome contribution! Let me quickly test it when I can get a chance before merging.

**Comment by niosus** [comment]
@pl-ca sure thing. Tell me if there is something wrong whenever you're ready :)

File: ClangAutoComplete.py [hunk 1]
```diff
@@ -11,53 +11,53 @@ class ClangAutoComplete(sublime_plugin.EventListener):
...
-	project_name_regex = re.compile("([^\.]+).sublime-project")
 	settings_time = 0
 
+	def to_bool(self, str_flag):
+		return str_flag == "true"
+
 	def load_settings(self):
 		# Only load the settings if they have changed
-		settings_modified_time = path.getmtime(sublime.packages_path()+"/ClangAutoComplete/"+"ClangAutoComplete.sublime-settings")
+		settings_modified_time = path.getmtime(
```
Comment (niosus) [inline ClangAutoComplete.py]: Hey @pl-ca I actually think there is a better way to deal with it, I just don't know how yet. We probably could only load the settings once the plugin gets reloaded (it happens automatically on settings change, I guess) and then we can just keep the settings inside the class variables. This probably won't work quite as expected given we have now moved to having settings in a separate folder and not in the repo

**Comment by pl-ca** [comment]
This cleanup improves the code quite a bit, but there is 1 thing I think we need to change. I'm not sure if/how I can update the merge request so I'll put my suggested patch in this comment.

I think this is needed because whenever those variables are not in use, the plugin throws an error.

```diff
@@ -27,11 +27,18 @@ class ClangAutoComplete(sublime_plugin.EventListener):

                settings = sublime.load_settings("ClangAutoComplete.sublime-settings")

+               project_path=""
+               project_name=""
+               file_parent_folder=""
+
                # Variable $project_base_path in settings will be replaced by sublime's project path
                variables = sublime.active_window().extract_variables()
-               project_path = variables['folder']
-               project_name = variables['project_base_name']
-               file_parent_folder = path.join(path.dirname(variables['file']), "..")
+               if 'folder' in variables:
+                       project_path = variables['folder']
+               if 'project_base_name' in variables:
+                       project_name = variables['project_base_name']
+               if 'file' in variables:
+                       file_parent_folder = path.join(path.dirname(variables['file']), "..")

                include_parent_folder = self.to_bool(settings.get("include_file_parent_folder"))
                self.complete_all = self.to_bool(settings.get("autocomplete_all"))
@@ -50,9 +57,8 @@ class ClangAutoComplete(sublime_plugin.EventListener):
                        include_dir = os.path.abspath(include_dir)

                if (self.verbose):
-                       print("project_base_name = {}".format(variables['project_base_name']))
-                       print("folder = {}".format(variables['folder']))
-                       print("file = {}".format(variables['file']))
+                       print("project_base_name = {}".format(project_path))
+                       print("folder = {}".format(project_name))
                        print("file_parent_folder = {}".format(file_parent_folder))

                if (include_parent_folder):
```

Otherwise I also tested on Windows and it worked great.

File: ClangAutoComplete.py (hunk 1 above)
Comment (pl-ca) [inline ClangAutoComplete.py]: You are right, that would be best however I'm not exactly sure how to do this... the sublime API is not always that clear :/. Last I tried, it did not automatically reload the plugin on settings change and that is why I had put that line there.

**Comment by niosus** [comment]
@pl-ca ah, yes, you're right. My bad, did not test properly. Will fix it in the merge.

**Comment by niosus** [comment]
@pl-ca ok, your suggestions are included in the last commit. In case there is something else - just ping me, or, alternatively, if the changes are small you could merge it and modify the code after the merge.

File: ClangAutoComplete.py (hunk 1 above)
Comment (niosus) [inline ClangAutoComplete.py]: It seems to reload the plugin every time I change the user settings, but you're right, I wouldn't bet my life on it.

**Pull request merged**